plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/BasicSQLDialect.java

```diff
@@ -213,7 +213,7 @@ public DBPIdentifierCase storesQuotedCase() {
     @NotNull
     @Override
     public String escapeString(String string) {
-        return string.replace("'", "''");
+        return string.replace("'", "''").replaceAll("\\\\(?![_%?])", "\\\\\\\\");

```

Comment:
It is definitely not a base logic.
I thought it is Snowflake specific?

This regex breaks mind. What does it do? If we need to escape back slash then we should do it much simpler.